mlir/lib/Dialect/Rock/utility/AmdArchDb.cpp

```diff
@@ -74,24 +83,207 @@ static constexpr AmdArchInfo
              /*maxSharedMemPerWG*/ 65536, /*numEUPerCU=*/4, /*minNumCU=*/36,
              /*hasFp8ConversionInstrs=*/false,
              /*hasOcpFp8ConversionInstrs=*/false, /*maxNumXCC=*/1),
-    gfx11Info(GemmFeatures::dot | GemmFeatures::atomic_add |
+    rdna3Info(GemmFeatures::dot | GemmFeatures::atomic_add |
                   GemmFeatures::atomic_fmax_f32 | GemmFeatures::wmma,
-              /*waveSize=*/32, /*maxWavesPerEU*/ 20, /*totalSGPRPerEU*/ 512,
+              /*waveSize=*/32, /*maxWavesPerEU*/ 16, /*totalSGPRPerEU*/ 800,
+              /*totalVGPRPerEU*/ 1536, /*totalSharedMemPerCU*/ 131072,
+              /*maxSharedMemPerWG*/ 65536, /*numEUPerCU=*/4, /*minNumCU=*/12,
+              /*hasFp8ConversionInstrs=*/false,
+              /*hasOcpFp8ConversionInstrs=*/false, /*maxNumXCC=*/1),
+    rdna4Info(GemmFeatures::dot | GemmFeatures::atomic_add |
+                  GemmFeatures::atomic_fmax_f32 | GemmFeatures::wmma |
+                  GemmFeatures::atomic_add_f16 | GemmFeatures::atomic_add_bf16,
+              /*waveSize=*/32, /*maxWavesPerEU*/ 16, /*totalSGPRPerEU*/ 800,
               /*totalVGPRPerEU*/ 1536, /*totalSharedMemPerCU*/ 131072,
               /*maxSharedMemPerWG*/ 65536, /*numEUPerCU=*/4, /*minNumCU=*/12,
               /*hasFp8ConversionInstrs=*/false,
-              /*hasOcpFp8ConversionInstrs=*/false, /*maxNumXCC=*/1);
+              /*hasOcpFp8ConversionInstrs=*/true, /*maxNumXCC=*/1);
+
+namespace {
+
+template <typename LHS, typename RHS>
+std::enable_if_t<std::is_assignable_v<LHS &, RHS &&>, void>
+checkAndSetInfo(StringRef name, LHS &lhs, RHS &&rhs) {
+  if (lhs != static_cast<LHS>(rhs)) {
+    LLVM_DEBUG(llvm::dbgs() << "NOTE: Value discrepancy for " << name << ": "
+                            << lhs << " (old) != " << rhs
+                            << " (new). Proceeding with " << rhs << ".\n");
+    lhs = std::forward<RHS>(rhs);
+  }
+}
+
+GemmFeatures &operator|=(GemmFeatures &lhs, const GemmFeatures &rhs) {
+  lhs = lhs | rhs;
+  return lhs;
+}
+
+GemmFeatures &operator&=(GemmFeatures &lhs, const GemmFeatures &rhs) {
+  lhs = lhs & rhs;
+  return lhs;
+}
```

Comment:
GemmFeatures already has `bitEnumClear` and `bitEnumSet` methods that you can use